src/main/java/org/benetech/mathshare/controllers/ProblemController.java

```diff
@@ -0,0 +1,41 @@
+package org.benetech.mathshare.controllers;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.benetech.mathshare.model.dto.ProblemSetDTO;
+import org.benetech.mathshare.service.ProblemService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/set")
+@SuppressFBWarnings(value = "SLF4J_FORMAT_SHOULD_BE_CONST",
+        justification = "We need Logger to contain not-constant data to be valuable")
+public class ProblemController {
+
+    private final Logger logger = LoggerFactory.getLogger(ProblemController.class);
+
+    @Autowired
+    private ProblemService problemService;
+
+    @GetMapping("/view/{code}")
+    ResponseEntity<ProblemSetDTO> getProblemSet(@PathVariable String code) {
+        try {
+            ProblemSetDTO body = problemService.findProblemsByUrlCode(code);
+            if (body != null) {
+                return new ResponseEntity<>(body, HttpStatus.OK);
+            } else {
+                logger.error("ProblemSet with code " + code + " wasn't found");
```

Comment:
Please, try to use String.format

Reply:
using String.format will still cause the SpotBugs violations, and I think current form is more readable, isn't it?

Reply:
I think there is a possibility to use internal string formatter available in log4j. IMO messages with {} are more readable especially when there are many vars:
https://stackoverflow.com/questions/30975026/log4j-implicit-string-formatting#30976958

Reply:
Ok, it might be more readable for many vars, so I can change it here to establish a convention. But it still causes the violation, as you can see here: https://github.com/KengoTODA/findbugs-slf4j/blob/master/README.md